extensions/flyway/runtime/src/main/java/io/quarkus/flyway/FlywayDataSource.java

```diff
@@ -0,0 +1,64 @@
+package io.quarkus.flyway;
+
+import static java.lang.annotation.ElementType.FIELD;
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.ElementType.PARAMETER;
+import static java.lang.annotation.ElementType.TYPE;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import javax.enterprise.util.AnnotationLiteral;
+import javax.inject.Named;
+import javax.inject.Qualifier;
+
+/**
+ * Qualifier used to specify which datasource will be used and therefore which flyway instance will be injected.<p<
+ * 
+ * Flyway instance can also be qualified by name using @{@link Named}.
+ * The name is the datasource name followed by "_flyway".
+ * 
+ * @author JohT
+ */
+@Target({ METHOD, FIELD, PARAMETER, TYPE })
+@Retention(RUNTIME)
+@Documented
+@Qualifier
+public @interface FlywayDataSource {
+
+    String value();
+
+    /**
+     * Supports inline instantiation of the {@link FlywayDataSource} qualifier.
+     *
+     * @author JohT
+     */
+    public static final class Literal extends AnnotationLiteral<FlywayDataSource> implements FlywayDataSource {
```

Comment:
I'm not sure this should be API. @mkouba WDYT?

Reply:
My reference for that is: "javax.enterprise.inject.Any"

Reply:
It's ok ;-).